File: code/java/start/src/FibTime.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.StandardOpenOption;
import java.util.Arrays;
import java.util.List;

public class FibTime {
    public static void main(String[] args) throws IOException {
        List<Integer> ns = Arrays.asList(20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30);
        Path p = Paths.get("fib_java.txt");
        Files.deleteIfExists(p);
        Fibonacci f = new Fibonacci();
        for (int n : ns) {
            long start = System.currentTimeMillis();
            f.fibonacci(n);
            long end = System.currentTimeMillis();
            Files.writeString(p, String.format("%d  %d\n", n, end - start),
                    StandardOpenOption.CREATE,
                    StandardOpenOption.APPEND);
        }
    }
}

File: code/python/start/fib_time.py
import time
from fibonacci import fibonacci


if __name__ == '__main__':
    ns = [20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30]
    with open('fib_python.txt', 'w') as f:
        for n in ns:
            start = int(round(time.time() * 1000))
            fibonacci(n)
            end = int(round(time.time() * 1000))
            f.write(f'{n}   {end - start}\n')
